contracts/partials/strategy/helpers.ligo

```diff
@@ -0,0 +1,86 @@
+function check_strategy_bounds(
+  const reserves        : nat;
+  const desired_reserves: nat;
+  const desired_rate_f  : nat;
+  const delta_rate_f    : nat)
+                        : unit is
+  block {
+    const rate_f = desired_reserves * Constants.precision / reserves;
+    const in_upper_bound = rate_f < (desired_rate_f + delta_rate_f);
+    const in_lower_bound = rate_f > nat_or_error(desired_rate_f - delta_rate_f, Errors.Math.nat_error);
+  } with require(in_upper_bound and in_lower_bound, Errors.Strategy.out_of_delta_bounds)
```

Comment:
It looks like the execution shouldn't fail because then the rebalance will never happen. It should be done if the  abs(rate_f) > delta_rate_f